### PR TITLE
🏗  Separate closure warning levels for compile vs. type check

### PR DESCRIPTION
**Pull request by rsimha**
Follow up to #23242

### CODE REVIEW

File: build-system/compile/compile.js [hunk 1]
```diff
@@ -328,8 +322,13 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
         'constantProperty',
         'globalThis'
       );
+      compilerOptions.jscomp_warning.push('accessControls');
```
Comment (erwinmombay) [inline build-system/compile/compile.js]: can we add a comment somewhere linking to the explanation of these error/warning groups.

File: build-system/compile/compile.js [hunk 2]
```diff
@@ -328,8 +322,13 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
...
+      compilerOptions.jscomp_off.push('moduleLoad', 'unknownDefines');
       compilerOptions.conformance_configs =
         'build-system/conformance-config.textproto';
+    } else {
+      compilerOptions.jscomp_warning.push('accessControls', 'moduleLoad');
```
Comment (erwinmombay) [inline build-system/compile/compile.js]: they should be on by default or no?

File: build-system/compile/compile.js (hunk 1 above)
Comment (rsimha) [reply build-system/compile/compile.js]: Good idea. Done.

File: build-system/compile/compile.js (hunk 2 above)
Comment (rsimha) [reply build-system/compile/compile.js]: Defaults are different for each type. This makes both of them warnings, which is what we want.

**Comment by rsimha** [comment]
@erwinmombay PTAL.

**Pull request merged**